### ISSUE TITLE
Reduce的作用是啥

### ISSUE THREAD

**Issue by xdnjust** [hquzhuguofeng/New-Pointer-Generator-Networks-for-Summarization-Chinese#6]
大神你好，我是新手，最近在学习你的代码，请问一下Reduce类的作用是什么，对应论文哪个功能，谢谢！

  class Reduce(nn.Module):
    
    def __init__(self, config):
        super(Reduce, self).__init__()
        self.config = config
        self.reduce_h = nn.Linear(config.hidden_dim * 2, config.hidden_dim)
        self.reduce_c = nn.Linear(config.hidden_dim * 2, config.hidden_dim)
        self.dropout = nn.Dropout(p=config.hidden_dropout_prob)

    def forward(self, h, c):
        assert self.config.hidden_dim == h.shape[2]
        assert self.config.hidden_dim == c.shape[2]
        h = h.reshape(-1, self.config.hidden_dim * 2)
        c = c.reshape(-1, self.config.hidden_dim * 2)
        h_output = self.dropout(self.reduce_h(h))
        c_output = self.dropout(self.reduce_c(c))
        h_output = F.relu(h_output)
        c_output = F.relu(c_output)
        # h_output.shape = c_output.shape = (batch,hidden)
        return h_output.unsqueeze(0), c_output.unsqueeze(0)
